pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -369,6 +426,111 @@ func (r *MachineSetSyncReconciler) updateSynchronizedConditionWithPatch(ctx cont
 	return nil
 }
 
+// createOrUpdateCAPIInfraMachineTemplate creates a CAPI infra machine template from a MAPI machine set, or updates if it exists and it is out of date.
+func (r *MachineSetSyncReconciler) createOrUpdateCAPIInfraMachineTemplate(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, infraMachineTemplate client.Object, newCAPIInfraMachineTemplate client.Object) (ctrl.Result, error) { //nolint:unparam
```

Comment:
It is because in the returned `(ctrl.Result, error)`
we always come back with `return ctrl.Result{}, createErr` so the Result is always nil and golanci-lint complains about it.
I'm under the impression that every time we define a method for the reconciler receiver, it is wise to return `(ctrl.Result, error)` as the returning values. Hence I haven't removed `ctrl.Result` but put the no-lint instead.